types/jsonld/index.d.ts

```diff
@@ -0,0 +1,217 @@
+        skipExpansion?: boolean;
+        format?: MimeNQuad;
+        produceGeneralizedRdf?: boolean;
+    }
+
+    // TODO Complete and uncomment if needed (see comments at the end of the file)
+    /* NOT USED AT THE MOMENT
+    // type Link = Common;
+    interface Issuer {
+
+        issuer?: IdentifierIssuer;   // a jsonld.IdentifierIssuer to use to label blank nodes.
+    }
+
+    type CreateNodeMap = Common&Issuer;
+
+    interface Merge extends Common, Issuer{
+
+        mergeNodes?: boolean;    //true to merge properties for nodes with the same ID,
+        //false to ignore new properties for nodes with the same ID once
+        //the ID has been defined; note that this may not prevent merging
+        //new properties where a node is in the `object` position
+        //(default: true).
+    }
+
+    interface Get {
+        documentLoader?: DocLoader;  // the document loader to use.
+    }
+
+    type ProcessContext = DocLoader;
+    */
+}
+
+export function compact(input: Document, ctx: Context, options: Options.Compact, callback: DocCallback): void;
+export function compact(input: Document, ctx: Context, callback: DocCallback): void;
+export function compact(input: Document, ctx: Context, options?: Options.Compact): Promise<JsonLd>;
+
+export function expand(input: Document, options: Options.Expand, callback: DocCallback): void;
+export function expand(input: Document, callback: DocCallback): void;
+export function expand(input: Document, options?: Options.Expand): Promise<JsonLd>;
+
+export function flatten(input: Document, ctx: Context, options: Options.Flatten, callback: DocCallback): void;
+export function flatten(input: Document, ctx: Context, callback: DocCallback): void;
+export function flatten(input: Document, ctx: Context, options?: Options.Flatten): Promise<JsonLd>;
+
+export function frame(input: Document, frame: Document, options: Options.Frame, callback: DocCallback): void;
+export function frame(input: Document, frame: Document, callback: DocCallback): void;
+export function frame(input: Document, frame: Document, options?: Options.Frame): Promise<JsonLd>;
+
+export function normalize(input: Document, options: Options.Normalize, callback: DocCallback): void;
+export function normalize(input: Document, callback: DocCallback): void;
+export function normalize(input: Document, options?: Options.Normalize): Promise<JsonLd>;
+
+export function fromRDF(dataset: RdfOrString, options: Options.FromRdf, callback: DocCallback): void;
+export function fromRDF(dataset: RdfOrString, callback: DocCallback): void;
+export function fromRDF(dataset: RdfOrString, options?: Options.FromRdf): Promise<JsonLd>;
+
+export function toRDF(input: Document, callback: Callback<RdfDataSet>): void;
+export function toRDF(input: Document, options: Options.ToRdf, callback: Callback<RdfDataSet>): void;
+export function toRDF(input: Document, options?: Options.ToRdf): Promise<RdfDataSet>;
+
+export let JsonLdProcessor: JsonLdProcessor;
```

Comment:
This is incorrect. What is exported is not a `JsonLdProcessor`-typed value, but a `JsonLdProcessor` class: https://github.com/digitalbazaar/jsonld.js/blob/master/lib/JsonLdProcessor.js#L7

Reply:
The class is just having the 3 static functions "expand", "flatten" and "compact". If I use a class instead of an interface the linter is complaining about that the class has only static properties. In my opinion it is correct to use the interface here. I have added an according section to the test file, to test if it is working.

Reply:
Should I change something anyway or can I mark it as resolved?